### ISSUE TITLE
make sidebar re-sizeable

### ISSUE THREAD

**Issue by joshinils** [OSM-de/osm-org-greasemonkey#5]
if possible somehow, make the width of the sidebar containing info on a way/node/note/changeset etc. re-sizeable so it is easier to read.

especially for notes with a longer conversation, it be better if the text is not contained in a (relatively) small margin on the left.

**Comment by joshinils** [comment]
oh, it's a simple button to toggle twice the width.
that should work for hard to read note threads, or long tags.

but i did think something along this could work better:
```css
#sidebar
{
    min-width: 350px;
    resize: horizontal;
    overflow: auto;
    max-height: fit-content;
    max-width: fit-content;
    /**border: 1px solid black;*/
}

#sidebar_content{
    width: unset !important;
}
```

**Comment by joshinils** [comment]
When first clicking the resize button it does nothing but add `style="width: 350px;"` to the `#sidebar` and `#sidebar_content` elements, only clicking it a second time resizes the sidebar.

The buttons also still only appear after moving the map.

**Comment by kmpoppe** [comment]
You're implying I can completely overwrite the CSS of the sidebar, I've still to see if that's at all possible.

The buttons only appearing after a map move was fixed at some point but seems to have keeled over 🤷, will have to look at that again.

**Comment by joshinils** [comment]
> You're implying I can completely overwrite the CSS of the sidebar, I've still to see if that's at all possible.

yes, i do that with https://addons.mozilla.org/de/firefox/addon/styl-us/

**Comment by kmpoppe** [comment]
> yes, i do that with https://addons.mozilla.org/de/firefox/addon/styl-us/

Or I can try to use `.cssText` so that this script doesn't have to rely on yet another addon.

**Comment by joshinils** [comment]
resizing works great, however, when moving the map the `modifyContent` gets called, and the css gets overwritten, meaning some `width: 897px` which gets added by re-sizing the sidebar gets lost, meaning the sidebar snaps to its original size any time one moves the map.

so adding the css should only happen once, if possible, otherwise any changed state gets lost when dragging the map.

**Comment by kmpoppe** [comment]
> resizing works great, however, when moving the map the `modifyContent` gets called, and the css gets overwritten, meaning some `width: 897px` which gets added by re-sizing the sidebar gets lost, meaning the sidebar snaps to its original size any time one moves the map.
> 
> so adding the css should only happen once, if possible, otherwise any changed state gets lost when dragging the map.

Fixed with f372989

**Comment by joshinils** [comment]
ugh, now there is always a white sidebar even when there is nothing in it.

well after removing these lines:
https://github.com/OSM-de/osm-org-greasemonkey/blob/f372989a76b24625b23945c2fa94d38e370bf06b/osm-org-greasemonkey.user.js#L38-L39
https://github.com/OSM-de/osm-org-greasemonkey/blob/f372989a76b24625b23945c2fa94d38e370bf06b/osm-org-greasemonkey.user.js#L44
it looks as before and works when nothing is in the sidebar.

**Comment by kmpoppe** [comment]
Yeah, I saw that as well but didn't have to chance to look further into it, thanks for supplying a fix.

**Comment by kmpoppe** [comment]
fixed with f372989a76b24625b23945c2fa94d38e370bf06b
